pyomo/contrib/fbbt/fbbt.py

```diff
@@ -1084,7 +970,77 @@ def _check_and_reset_bounds(var, lb, ub):
     return lb, ub
 
 
-class _FBBTVisitorLeafToRoot(ExpressionValueVisitor):
+def _before_constant(visitor, child):
+    if child in visitor.bnds_dict:
+        pass
+    else:
+        visitor.bnds_dict[child] = (child, child)
+    return False, None
+
+
+def _before_var(visitor, child):
+    if child in visitor.bnds_dict:
+        return False, None
+    elif child.is_fixed() and not visitor.ignore_fixed:
+        lb = value(child.value)
+        ub = lb
+    else:
+        lb = value(child.lb)
+        ub = value(child.ub)
+        if lb is None:
+            lb = -interval.inf
+        if ub is None:
+            ub = interval.inf
+        if lb - visitor.feasibility_tol > ub:
+            raise InfeasibleConstraintException(
+                'Variable has a lower bound that is larger than its '
+                'upper bound: {0}'.format(str(child))
+            )
+    visitor.bnds_dict[child] = (lb, ub)
+    return False, None
+
+
+def _before_NPV(visitor, child):
+    if child in visitor.bnds_dict:
+        return False, None
+    val = value(child)
+    visitor.bnds_dict[child] = (val, val)
+    return False, None
+
+
+def _before_other(visitor, child):
```

Comment:
What is this handler for?

Reply:
The previous walker basically had a default that if it didn't understand what was going on, it still returned bounds of `(-inf, inf)`. So this maintains that behavior. I'm not sure that it's a good idea long-term though?

Reply:
I think we should, but that's actually going to require a more significant rewrite because right now this is catching both things we don't understand and things that we do understand, just don't need a beforeChild handler for (like SumExpression). So I think it would make the most sense to move this onto the new BeforeChildDispatcher base class and then finally get pedantic about when we're confused and when we just don't care. But I'd rather that be a separate PR.